includes/discovery/ipv6-addresses.inc.php

```diff
@@ -41,13 +41,18 @@
             $ipv6_prefixlen = explode('.', $ipv6_prefixlen);
             $ipv6_prefixlen = str_replace('"', '', end($ipv6_prefixlen));
 
+            if (Str::contains($ipv6_prefixlen, 'SNMPv2-SMI::zeroDotZero')) {
+                d_echo('Incomplete IPv6 data in IF-MIB');
+                $oids = trim(Str::replaceFirst($data, '', $oids));
```

Comment:
what is the point of this? would a continue work?

Reply:
Not alone.

What I'm trying to do here is detect the case of 'IP-MIB V6 entries are broken on this device _completely_' vs 'IP-MB doesn't contain entries for _some_ addresses for $reasons', without creating a mess of conditions or storing additional state.

The existing code flips from IP-MIB to IPV6-MIB based on $oids being empty, so this block gradually empties $oids so the existing condition fires.

To use a continue here,  you'd need to store the results of ipv6_discovery_process and compare the number of positive results against the number of line breaks in $oid, then add an additional condition that the alternate code runs if the counts differ.

The whole logic of the discovery process is backwards IMO - IPV6-MIB should be looked at first, but that's an extensive refactor and there isn't the test data available to support it right now.

Reply:
I'm also not a fan of all the crazy parsing of the snmpwalk output, could be done better.  I'm probably fine with this, but it is extremely hard to understand what it is doing...

Reply:
It's just awful all round.

prefixlen at this point on JunOS does not contain a prefix length, it ends up containing the whole string captured by snmpget with the zeroDotZero data type, rather than an 'oid not found', 'no object at this id' or similar error.

zeroDotZero doesn't seem to be well documented, but it seems it should be considered as a 'null' value.

$oids doesn't contain OIDs, it contains new line seperated pairs of IPv6 addresses and ifIndexes.